crates/hyperswitch_connectors/src/connectors/aci/transformers.rs

```diff
@@ -94,6 +94,38 @@ impl TryFrom<&ConnectorAuthType> for AciAuthType {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum AciRecurringType {
+    Initial,
+    Repeated,
+    RegistrationBased,
+}
+
+fn get_str(key: &str, metadata: &serde_json::Value) -> Option<String> {
+    metadata
+        .get(key)
+        .and_then(|v| v.as_str())
+        .map(|s| s.to_string())
+}
+
+fn determine_recurring_type_for_3ds(
+    item: &PaymentsAuthorizeRouterData,
+) -> Option<AciRecurringType> {
+    if let Some(metadata) = item.connector_meta_data.as_ref() {
+        let metadata_value = metadata.peek();
+        if let Some(recurring_type) = get_str("recurring_type", metadata_value) {
+            return match recurring_type.as_str() {
```

Comment:
can we use to_lowercase before the match?  it’s a good to normalize the string before matching